modules/build_environment_module.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+"""
+This program is used for deploying code modules onto the NCI.
+
+It is configured by a YAML file, which specifies:
+ - variables
+ - files to copy with permissions
+ - template files with their destination (eg, modulefiles)
+ - (opt) Conda environment to create
+ - (opt) Pip style requirements.txt to install to a directory
+
+It requires python 3.6+ and pyyaml. To run it on raijin at the NCI:
+
+  module load python3/3.6.2
+  pip install --user pyyaml
+  ./build_environment_module.py dea-env/modulespec.yaml
+
+It used to be able to perform a miniconda installation, but that turned out to
+be flaky, so we now maintain a central miniconda install, and create environments
+as an where required. With the added benefit of keeping a central cache of
+packages.
+"""
+
+
+import datetime
+import os
+import subprocess
+import sys
+import tempfile
+import urllib.request
+from pathlib import Path
+import shutil
+import string
+import logging
+import yaml
+
+MODULE_DIR = '/g/data/v10/public/modules'
+
+LOG = logging.getLogger('environment_module_builder')
+
+def pre_check(config):
+    LOG.debug('Performing pre-check before installing module')
+    if "PYTHONPATH" in os.environ:
+        raise Exception("The PYTHONPATH environment variable must NOT be set when creating modules.")
+
+    module_path = Path(config['variables']['module_path'])
+    if module_path.exists():
+        raise Exception(f"The destination path {module_path} already exists, "
+                        f"please remove it and try again.")
+
+
+def prep(config_path):
+    LOG.debug('Preparing environment variables')
+    # Write files as group and world readable
+    os.umask(0o22)
+    os.chdir(config_path.parent)
+    os.environ['LC_ALL'] = 'en_AU.utf8'
+    os.environ['LANG'] = 'C.UTF-8'
+
+
+def date(date_format="%Y%m%d") -> str:
+    return datetime.datetime.now().strftime(date_format)
+
+
+def make_output_dir(base_path) -> Path:
+    today = date()
+
+    output_dir = base_path / today
+
+    while output_dir.exists():
+        index = 1
```

Comment:
`+= 1` ?

Reply:
This function isn't used at the moment, it'll just fail if you attempt to create a two versions of a module on the same day. Simple solution is to delete the first one, or wait until tomorrow.

This was going to automatically generate a new `YYYYMMDD_{index}`, but it started feeling complicated and unnecessary.

Will probably just delete this function.